Gtest/AudioClient.cpp

```diff
@@ -0,0 +1,160 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_TAG "AudioClient"
+#include "AudioClient.h"
+#include <log/log.h>
+
+using namespace std::chrono_literals;
+using namespace std;
+using namespace vhal::client::audio;
+using namespace vhal::client;
+
+int AudioClient::startDummyStreamerForInput()
+{
+    string       ip_addr("172.100.0.2");
+    uint16_t     port_num(18767);
+    resetCmdFlagsForAudioSink();
+    TcpConnectionInfo conn_info = { ip_addr, port_num };
+    AudioSink audio_sink(conn_info,[&](const CtrlMessage& ctrl_msg) {
+        switch (ctrl_msg.cmd) {
+            case Command::kOpen: {
+                is_open_for_sink = true;
+                ALOGV("Received Open command from Audio VHal");
+                break;
+            }
+            case Command::kClose: {
+                is_close_for_sink = true;
+                ALOGV("Received Close command from VHal");
+                break;
+            }
+            default:
+                ALOGE("Unknown Command received, exiting with failure");
+                exit(1);
+        }
+    });
+    ALOGI("Waiting Audio Open callback For Input..");
+
+    // Need to be alive :)
+    while (is_running_sink) {
+        this_thread::sleep_for(5ms);
+    }
+    return 0;
+}
+
+int AudioClient::startDummyStreamerForOutput() {
+    string       ip_addr("172.100.0.2");
+    uint16_t     port_num(18768);
+    resetCmdFlagsForAudioSource();
+    TcpConnectionInfo conn_info = { ip_addr, port_num };
+    AudioSource audio_source(conn_info, [&](const CtrlMessage& ctrl_msg) {
+        switch (ctrl_msg.cmd) {
+        case Command::kOpen:{
+            is_open_for_source = true;
+            ALOGV("Received Open command from Audio VHal");
+            break;
+        }
+        case Command::kStartstream:{
+            is_stream_started = true;
+            ALOGV("Received Startstream command from Audio VHal");
+	    break;
+	}
+        case Command::kData:{
+            is_data_received = true;
+            ALOGV("Received Data command from Audio VHal");
+            break;
+        }
+        case Command::kStopstream:{
+            is_stream_stopped = true;
+            ALOGV("Received Stopstream command from Audio VHal");
+	    break;
+	}
+        case Command::kClose:{
+            is_close_for_source = true;
+            ALOGV("Received Close command from VHal");
+	    break;
+	}
+        default:
+            ALOGE("Unknown Command received, exiting with failure");
+            exit(1);
+        }
+    });
+    ALOGI("Waiting Audio Open callback For Output...");
+    // Need to be alive :)
+    while (is_running_source) {
+        this_thread::sleep_for(5ms);
+    }
+
+    return 0;    
+}
+
+void AudioClient::stopDummyStreamerForInput()
```

Comment:
like rest of the file, can we have the { in the same line as function definition?